Sources/App/AppDelegate.swift

```diff
@@ -201,10 +201,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         configurationForConnecting connectingSceneSession: UISceneSession,
         options: UIScene.ConnectionOptions
     ) -> UISceneConfiguration {
-        let activity = options.userActivities
-            .compactMap { SceneActivity(activityIdentifier: $0.activityType) }
-            .first ?? .webView
-        return activity.configuration
+        if #available(iOS 16.0, *), connectingSceneSession.role == UISceneSession.Role.carTemplateApplication {
+            let scene = UISceneConfiguration(name: "CarPlay", sessionRole: connectingSceneSession.role)
```

Comment:
this should be done in the Info.plist (like other scene configurations) and through the SceneActivity type (which handles specifying it.

feel free to give it a different initializer which handles role if that's the differentiating thing.